src/Movie/Movie.js

```diff
@@ -56,6 +57,12 @@ class Movie extends Component {
         this.viewMovie(this.props.id)
     }
 
+    componentDidUpdate() {
+        if (this.state.location !== this.props.location) {
+            this.viewMovie(this.props.location)
```

Comment:
When the component updates (every time the App passes a fresh prop of location) the Movie checks if it should fetch another movie by comparing the current, and new `location` value.